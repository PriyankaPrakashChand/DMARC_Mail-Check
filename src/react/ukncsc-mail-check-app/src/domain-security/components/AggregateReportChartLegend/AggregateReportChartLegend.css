.ChartLegend {
  list-style-type: none;
  padding: 0px 0px 50px 0px;
}

@media only screen and (max-width: 767px) {
  .ChartLegend {
    list-style-type: none;
    padding: 0px 0px 0px 0px;
    margin: 0px 0px 0px 0px;
  }
}

.ChartLegendHeader {
  display: inline-block;
  margin-right: 5px;
}

.ChartLegendMarker {
  display: inline-block;
  height: 16px;
  width: 16px;
  border: 2px solid;
  margin: 0px 5px 0px 5px;
}
