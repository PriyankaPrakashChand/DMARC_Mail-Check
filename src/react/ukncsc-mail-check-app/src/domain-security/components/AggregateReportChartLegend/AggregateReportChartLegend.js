import React from 'react';
import { Header } from 'semantic-ui-react';

import './AggregateReportChartLegend.css';

export default ({ children }) => (
  <ul className="ChartLegend">
    {children.map(({ name, background, stroke }) => (
      <li key={name}>
        <Header as="h2" className="ChartLegendHeader">
          -
        </Header>
        <div className="ChartLegendMarker" style={{ background, stroke }} />
        {name}
      </li>
    ))}
  </ul>
);
