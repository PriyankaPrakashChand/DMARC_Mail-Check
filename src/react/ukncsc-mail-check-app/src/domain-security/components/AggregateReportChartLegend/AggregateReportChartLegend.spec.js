import React from 'react';
import { render } from 'react-testing-library';
import AggregateReportChartLegend from './AggregateReportChartLegend';

describe('AggregateReportChartLegend', () => {
  let container;

  describe('when getting a chart legend', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportChartLegend>
          {[
            { name: 'item 1', background: '#fff000', stroke: '#ffffff' },
            { name: 'item 2', background: '#ffff00', stroke: '#fffff0' },
          ]}
        </AggregateReportChartLegend>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
