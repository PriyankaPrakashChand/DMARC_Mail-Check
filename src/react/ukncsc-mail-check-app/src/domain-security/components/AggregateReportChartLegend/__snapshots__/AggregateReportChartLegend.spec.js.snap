// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateReportChartLegend when getting a chart legend it should match the snapshot 1`] = `
<div>
  <ul
    class="ChartLegend"
  >
    <li>
      <h2
        class="ui header ChartLegendHeader"
      >
        -
      </h2>
      <div
        class="ChartLegendMarker"
        style="background: rgb(255, 240, 0); stroke: #ffffff;"
      />
      item 1
    </li>
    <li>
      <h2
        class="ui header ChartLegendHeader"
      >
        -
      </h2>
      <div
        class="ChartLegendMarker"
        style="background: rgb(255, 255, 0); stroke: #fffff0;"
      />
      item 2
    </li>
  </ul>
</div>
`;
